src/gtfs/read.rs

```diff
@@ -373,22 +337,56 @@ pub fn read_stops<P: AsRef<path::Path>>(
     Ok((stopareas, stoppoints))
 }
 
-#[derive(Serialize, Deserialize, Debug)]
-pub struct Config {
-    pub contributor: objects::Contributor,
-    pub dataset: objects::Dataset,
+#[derive(Deserialize, Debug)]
+struct Dataset {
+    dataset_id: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct Config {
+    contributor: objects::Contributor,
+    dataset: Dataset,
 }
 
-pub fn read_config(
-    config: Config,
+pub fn read_config<P: AsRef<path::Path>>(
+    config_path: Option<P>,
 ) -> Result<
     (
         CollectionWithId<objects::Contributor>,
         CollectionWithId<objects::Dataset>,
     ),
 > {
-    let contributors = CollectionWithId::new(vec![config.contributor])?;
-    let datasets = CollectionWithId::new(vec![config.dataset])?;
+    let contributor;
+    let dataset;
+    if let Some(config_path) = config_path {
+        let json_config_file = File::open(config_path)?;
+        let config: Config = serde_json::from_reader(json_config_file)?;
+        info!("config loaded: {:#?}", config);
+
+        contributor = config.contributor;
+
+        use chrono::{Duration, Utc};
+        let duration = Duration::days(15);
+        let today = Utc::today();
+        let start_date = today - duration;
+        let end_date = today + duration;
+        dataset = objects::Dataset {
```

Comment:
I don't get it, dataset is not given in the config?

Reply:
dataset in the config only have dataset_id and optional properties
see https://github.com/CanalTP/navitia_model/pull/57/files#diff-a3fce5151b3ec922f7e1e2ed742d4ea2R28